### ISSUE TITLE
Test failures with Python 3.9

### ISSUE THREAD

**Issue by hroncok** [serge-sans-paille/gast#46]
With Python 3.9.0b4, I get:

```
$ tox -e py39
GLOB sdist-make: .../gast/setup.py
py39 inst-nodeps: .../gast/.tox/.tmp/package/1/gast-0.3.3.zip
py39 installed: apipkg==1.5,astunparse==1.6.3,attrs==19.3.0,execnet==1.7.1,gast==0.3.3,more-itertools==8.4.0,packaging==20.4,pep8==1.7.1,pluggy==0.13.1,py==1.9.0,pyparsing==2.4.7,pytest==5.4.3,pytest-cache==1.0,pytest-pep8==1.0.6,six==1.15.0,wcwidth==0.2.5
py39 run-test-pre: PYTHONHASHSEED='155329557'
py39 run-test: commands[0] | py.test --pep8
.../gast/.tox/py39/lib/python3.9/site-packages/pep8.py:110: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
============================= test session starts ==============================
platform linux -- Python 3.9.0b4, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
cachedir: .tox/py39/.pytest_cache
rootdir: .../gast
plugins: pep8-1.0.6
collected 45 items

setup.py .                                                               [  2%]
gast/__init__.py .                                                       [  4%]
gast/ast2.py .                                                           [  6%]
gast/ast3.py .                                                           [  8%]
gast/astn.py .                                                           [ 11%]
gast/gast.py .                                                           [ 13%]
tests/__init__.py .                                                      [ 15%]
tests/test_api.py .......F......                                         [ 46%]
tests/test_compat.py ..FFFFF.F...........                                [ 91%]
tests/test_self.py .F..                                                  [100%]

=================================== FAILURES ===================================
______________________ APITestCase.test_increment_lineno _______________________

self = <tests.test_api.APITestCase testMethod=test_increment_lineno>

    def test_increment_lineno(self):
        tree = gast.Constant(value=1, kind=None)
        tree.lineno = 1
>       gast.increment_lineno(tree)

tests/test_api.py:64: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

node = <gast.gast.Constant object at 0x7fec385de850>, n = 1

    def increment_lineno(node, n=1):
        """
        Increment the line number and end line number of each node in the tree
        starting at *node* by *n*. This is useful to "move code" to a different
        location in a file.
        """
        for child in walk(node):
            if 'lineno' in child._attributes:
                child.lineno = getattr(child, 'lineno', 0) + n
            if 'end_lineno' in child._attributes:
>               child.end_lineno = getattr(child, 'end_lineno', 0) + n
E               TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

gast/gast.py:379: TypeError
___________________________ CompatTestCase.test_Call ___________________________

self = <tests.test_compat.CompatTestCase testMethod=test_Call>

    def test_Call(self):
        self.maxDiff = None
        code = 'foo(x, y=1, *args, **kwargs)'
        tree = gast.parse(code)
>       compile(gast.gast_to_ast(tree), '<test>', 'exec')
E       TypeError: required field "lineno" missing from keyword

tests/test_compat.py:191: TypeError
_________________________ CompatTestCase.test_Ellipsis _________________________

self = <tests.test_compat.CompatTestCase testMethod=test_Ellipsis>

    def test_Ellipsis(self):
        self.maxDiff = None
        code = 'def foo(a): a[...]'
        tree = gast.parse(code)
        compile(gast.gast_to_ast(tree), '<test>', 'exec')
        norm = ("Module(body=[FunctionDef(name='foo', args=arguments(args=["
                "Name(id='a', ctx=Param(), annotation=None, type_comment=None)"
                "], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[]"
                ", kwarg=None, defaults=[]), body=[Expr(value=Subscript(value="
                "Name(id='a', ctx=Load(), annotation=None, type_comment=None)"
                ", slice=Index(value=Constant(value=Ellipsis, kind=None)), ctx"
                "=Load()))], decorator_list=[], returns=None, type_comment="
                "None)], type_ignores=[])")
>       self.assertEqual(gast.dump(tree), norm)
E       AssertionError: "Modu[300 chars]lice=Constant(value=Ellipsis, kind=None), ctx=[77 chars]=[])" != "Modu[300 chars]lice=Index(value=Constant(value=Ellipsis, kind[90 chars]=[])"
E       - Module(body=[FunctionDef(name='foo', args=arguments(args=[Name(id='a', ctx=Param(), annotation=None, type_comment=None)], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[], kwarg=None, defaults=[]), body=[Expr(value=Subscript(value=Name(id='a', ctx=Load(), annotation=None, type_comment=None), slice=Constant(value=Ellipsis, kind=None), ctx=Load()))], decorator_list=[], returns=None, type_comment=None)], type_ignores=[])
E       ?                                                                                                                                                                                                                                                                                                                                                         ^
E       + Module(body=[FunctionDef(name='foo', args=arguments(args=[Name(id='a', ctx=Param(), annotation=None, type_comment=None)], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[], kwarg=None, defaults=[]), body=[Expr(value=Subscript(value=Name(id='a', ctx=Load(), annotation=None, type_comment=None), slice=Index(value=Constant(value=Ellipsis, kind=None)), ctx=Load()))], decorator_list=[], returns=None, type_comment=None)], type_ignores=[])
E       ?                                                                                                                                                                                                                                                                                                                    ++++++++++++                                     ^^

tests/test_compat.py:303: AssertionError
_________________________ CompatTestCase.test_ExtSlice _________________________

self = <tests.test_compat.CompatTestCase testMethod=test_ExtSlice>

    def test_ExtSlice(self):
        code = 'def foo(a): a[:,:]'
        tree = gast.parse(code)
>       compile(gast.gast_to_ast(tree), '<test>', 'exec')
E       TypeError: required field "lineno" missing from expr

tests/test_compat.py:263: TypeError
_____________________ CompatTestCase.test_ExtSliceEllipsis _____________________

self = <tests.test_compat.CompatTestCase testMethod=test_ExtSliceEllipsis>

    def test_ExtSliceEllipsis(self):
        self.maxDiff = None
        code = 'def foo(a): a[1, ...]'
        tree = gast.parse(code)
        compile(gast.gast_to_ast(tree), '<test>', 'exec')
        norm = ("Module(body=[FunctionDef(name='foo', args=arguments(args=["
                "Name(id='a', ctx=Param(), annotation=None, type_comment=None)"
                "], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[]"
                ", kwarg=None, defaults=[]), body=[Expr(value=Subscript(value="
                "Name(id='a', ctx=Load(), annotation=None, type_comment=None)"
                ", slice=Index(value=Tuple(elts=[Constant(value=1, kind=None)"
                ", Constant(value=Ellipsis, kind=None)], ctx=Load())), ctx="
                "Load()))], decorator_list=[], returns=None, type_comment="
                "None)], type_ignores=[])")
>       self.assertEqual(gast.dump(tree), norm)
E       AssertionError: "Modu[300 chars]lice=Tuple(elts=[Constant(value=1, kind=None),[133 chars]=[])" != "Modu[300 chars]lice=Index(value=Tuple(elts=[Constant(value=1,[146 chars]=[])"
E       - Module(body=[FunctionDef(name='foo', args=arguments(args=[Name(id='a', ctx=Param(), annotation=None, type_comment=None)], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[], kwarg=None, defaults=[]), body=[Expr(value=Subscript(value=Name(id='a', ctx=Load(), annotation=None, type_comment=None), slice=Tuple(elts=[Constant(value=1, kind=None), Constant(value=Ellipsis, kind=None)], ctx=Load()), ctx=Load()))], decorator_list=[], returns=None, type_comment=None)], type_ignores=[])
E       ?                                                                                                                                                                                                                                                                                                                                                                                                                 ^
E       + Module(body=[FunctionDef(name='foo', args=arguments(args=[Name(id='a', ctx=Param(), annotation=None, type_comment=None)], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[], kwarg=None, defaults=[]), body=[Expr(value=Subscript(value=Name(id='a', ctx=Load(), annotation=None, type_comment=None), slice=Index(value=Tuple(elts=[Constant(value=1, kind=None), Constant(value=Ellipsis, kind=None)], ctx=Load())), ctx=Load()))], decorator_list=[], returns=None, type_comment=None)], type_ignores=[])
E       ?                                                                                                                                                                                                                                                                                                                    ++++++++++++                                                                                             ^^

tests/test_compat.py:319: AssertionError
________________________ CompatTestCase.test_ExtSlices _________________________

self = <tests.test_compat.CompatTestCase testMethod=test_ExtSlices>

    def test_ExtSlices(self):
        code = 'def foo(a): a[1,:]'
        tree = gast.parse(code)
>       compile(gast.gast_to_ast(tree), '<test>', 'exec')
E       TypeError: required field "lineno" missing from expr

tests/test_compat.py:278: TypeError
__________________________ CompatTestCase.test_Index ___________________________

self = <tests.test_compat.CompatTestCase testMethod=test_Index>

    def test_Index(self):
        code = 'def foo(a): a[1]'
        tree = gast.parse(code)
        compile(gast.gast_to_ast(tree), '<test>', 'exec')
        norm = ("Module(body=[FunctionDef(name='foo', args=arguments(args=["
                "Name(id='a', ctx=Param(), annotation=None, type_comment=None)"
                "], posonlyargs=[], vararg=None, kwonlyargs=[], kw_defaults=[]"
                ", kwarg=None, defaults=[]), body=[Expr(value=Subscript(value="
                "Name(id='a', ctx=Load(), annotation=None, type_comment=None)"
                ", slice=Index(value=Constant(value=1, kind=None)), ctx=Load()"
                "))], decorator_list=[], returns=None, type_comment=None)]"
                ", type_ignores=[])")
>       self.assertEqual(gast.dump(tree), norm)
E       AssertionError: "Modu[300 chars]lice=Constant(value=1, kind=None), ctx=Load())[70 chars]=[])" != "Modu[300 chars]lice=Index(value=Constant(value=1, kind=None))[83 chars]=[])"
E       Diff is 1563 characters long. Set self.maxDiff to None to see it.

tests/test_compat.py:258: AssertionError
___________________________ SelfTestCase.testCompile ___________________________

self = <tests.test_self.SelfTestCase testMethod=testCompile>

    def testCompile(self):
        for src_py in self.srcs:
            with open(src_py) as f:
                content = f.read()
            gnode = gast.parse(content)
>           compile(gast.gast_to_ast(gnode), src_py, 'exec')
E           TypeError: required field "lineno" missing from keyword

tests/test_self.py:26: TypeError
=============================== warnings summary ===============================
.tox/py39/lib/python3.9/site-packages/pytest_pep8.py:38: 10 tests with warnings
  .../gast/.tox/py39/lib/python3.9/site-packages/pytest_pep8.py:38: PytestDeprecationWarning: direct construction of Pep8Item has been deprecated, please use Pep8Item.from_parent
    return Pep8Item(path, parent, pep8ignore, config._max_line_length)

.tox/py39/lib/python3.9/site-packages/_pytest/nodes.py:254
  .../gast/.tox/py39/lib/python3.9/site-packages/_pytest/nodes.py:254: PytestUnknownMarkWarning: Unknown pytest.mark.pep8 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    marker_ = getattr(MARK_GEN, marker)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED tests/test_api.py::APITestCase::test_increment_lineno - TypeError: uns...
FAILED tests/test_compat.py::CompatTestCase::test_Call - TypeError: required ...
FAILED tests/test_compat.py::CompatTestCase::test_Ellipsis - AssertionError: ...
FAILED tests/test_compat.py::CompatTestCase::test_ExtSlice - TypeError: requi...
FAILED tests/test_compat.py::CompatTestCase::test_ExtSliceEllipsis - Assertio...
FAILED tests/test_compat.py::CompatTestCase::test_ExtSlices - TypeError: requ...
FAILED tests/test_compat.py::CompatTestCase::test_Index - AssertionError: "Mo...
FAILED tests/test_self.py::SelfTestCase::testCompile - TypeError: required fi...
================== 8 failed, 37 passed, 11 warnings in 0.36s ===================
ERROR: InvocationError for command .../gast/.tox/py39/bin/py.test --pep8 (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py39: commands failed
```

**Comment by hroncok** [comment]
The AssertionErrors seem to be caused by different output of `ast.dump()`. See:

- https://bugs.python.org/issue36287
- https://github.com/python/cpython/pull/18843
- https://github.com/simonpercivall/astunparse/issues/56

The `TypeError: required field "lineno" missing from keyword` is puzzling to me. It might be related to https://bugs.python.org/issue38131

**Comment by isidentical** [comment]
> The TypeError: required field "lineno" missing from keyword is puzzling to me. It might be related to https://bugs.python.org/issue38131

I think it is related with python/cpython#19283

**Comment by serge-sans-paille** [comment]
Should be fixed by 0.4.0 (?)

**Comment by hroncok** [comment]
> Should be fixed by 0.4.0 

It is. Sorry for the delayed answer.
